lssvr/lssvr.py

```diff
@@ -1,92 +1,150 @@
 
 
 class LSSVR(BaseEstimator, RegressorMixin):
-    def __init__(self, C=2, kernel='linear', gamma=None):
-        self.supportVectors      = None
-        self.supportVectorLabels = None
+    """Least Squares Support Vector Regression.
+
+    Parameters
+    ----------
+    C : float, default=2.0
+        Regularization parameter. The strength of the regularization is
+        inversely proportional to C. Must be strictly positive.
+
+    kernel : {'linear', 'rbf'}, default='linear'
+        Specifies the kernel type to be used in the algorithm.
+        It must be 'linear', 'rbf' or a callable.
+
+    gamma : float, default = None
+        Kernel coefficient for 'rbf'
+
+
+    Attributes
+    ----------
+    support_: boolean np.array of shape (n_samples,), default = None
+        Array for support vector selection.
+
+    alpha_ : array-like
+        Weight matrix
+
+    bias_ : array-like
+        Bias vector
+
+
+    """
+
+    def __init__(self, C=2.0, kernel='linear', gamma=None):
         self.C = C
+        self.kernel = kernel
         self.gamma = gamma
-        self.kernel= kernel
-        self.idxs  = None
-        self.K = None
-        self.bias = None 
-        self.alphas = None
-
-    def set_params(self, **parameters):
-        for parameter, value in parameters.items():
-            setattr(self, parameter, value)
-        return self
 
-    def fit(self, x_train, y_train):
-        # self.idxs can be used to select points as support vectors,
-        # so you need another algorithm or criteria to choose them
-        if type(self.idxs) == type(None):
-            self.idxs=np.ones(x_train.shape[0], dtype=bool)
+    def fit(self, X, y, support=None):
+        """Fit the model according to the given training data.
+        Parameters
+        ----------
+        X : {array-like, sparse matrix} of shape (n_samples, n_features)
+            Training data
+
+        y : array-like of shape (n_samples,) or (n_samples, n_targets)
+            Target values.
+
+        support : boolean np.array of shape (n_samples,), default = None
+            Array for support vector selection.
+
+        Returns
+        -------
+        self : object
+            An instance of the estimator.
+        """
+
+        X, y = check_X_y(X, y, multi_output=True, dtype='float')
 
-        self.supportVectors      = x_train[self.idxs, :]
-        self.supportVectorLabels = y_train[self.idxs]
+        if not support:
+            self.support_ = np.ones(X.shape[0], dtype=bool)
+        else:
+            self.support_ = check_array(support, ensure_2d=False, dtype='bool')
 
-        K = self.kernel_func(self.kernel, x_train, self.supportVectors, self.gamma)
+        self.support_vectors_ = X[self.support_, :]
+        support_labels = y[self.support_]
 
-        self.K = K
-        OMEGA = K
-        OMEGA[self.idxs, np.arange(OMEGA.shape[1])] += 1/self.C
+        self.K_ = self.kernel_func(X, self.support_vectors_)
+        omega = self.K_.copy()
```

Comment:
I made omega a deep copy, because otherwise the kernel was also being modified with the regularization. It should be that way or my intuition was right?

Reply:
Make sense. For me, this looks better.